### PR TITLE
pd-recovery: add the new way to recover

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Users can recover the PD from a PD node.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.4 (TiDB 6.4 versions)
- [x] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
https://github.com/tikv/pd/pull/4014
### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@nolouch Could you please invite a tech review?

File: pd-recover.md [hunk 1]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
 
-## 快速开始
+## 方式一：从剩余 PD 集群重建
 
-### 获取 Cluster ID
+PD 集群一旦超过大多数的节点发生灾难，则无法提供服务。当还有 PD 节点存活时，可以选择一个存活的 PD 节点强制修改 raft group 的成员为，从而使该节点恢复服务。具体方式为：
+
+### 第 1 步：停掉所有节点
+
+把集群中的服务进程都停止(TiKV/TIFlash/TiDB)，防止在还未恢复过程中与 PD 参数交互，造成数据错乱等无法挽救的异常状况。
```
Comment (lhy1024) [inline pd-recover.md:33]:
```suggestion
把集群中的服务进程都停止(TiKV/TiFlash/TiDB)，防止在还未恢复过程中与 PD 参数交互，造成数据错乱等无法挽救的异常状况。
```

File: pd-recover.md [hunk 2]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+{{< copyable "shell-regular" >}}
+
+```
+ ./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml 
+```
+
+### 第 3 步： 使用 pd-recover 修复元数据
+
+由于是存少数派节点直接恢复服务，该节点可能存在数据是落后的，特别是 alloc_id，tso 这种一旦发生回退，则可能会给集群数据错乱或不可用的影响。因此，我们需要通过 pd-recover 来修改一些元数据，从而使得这个 PD 能继续提供正确的分配 ID，TSO 等服务，具体命令参考：
```
Comment (lhy1024) [inline pd-recover.md:47]:
```suggestion
由于是从少数派节点直接恢复服务，该节点可能存在数据是落后的，特别是 alloc_id，tso 这种一旦发生回退，则可能会给集群数据错乱或不可用的影响。因此，我们需要通过 pd-recover 来修改一些元数据，从而使得这个 PD 能继续提供正确的分配 ID，TSO 等服务，具体命令参考：
```

File: pd-recover.md [hunk 3]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+### 第 4 步：重启这个 PD
+
+当出现上一步出现 `recovery is successful` 的提示信息时，重启整个集群。
+
+### 第 5 步：扩容 PD，拉起集群
+
+通过部署工具扩容 PD，并把其他组件都拉起来。至此服务恢复
+
+## 方式二：完全重建 PD 集群
```
Comment (lhy1024) [inline pd-recover.md:69]: 是否需要“方式三：结合 online recovery 恢复多中心集群”

File: pd-recover.md [hunk 4]
```diff
@@ -116,7 +160,7 @@ cat {{/path/to}}/pd*.log | grep "idAllocator allocates a new id" |  awk -F'=' '{
 ./pd-recover -endpoints http://10.0.1.13:2379 -cluster-id 6747551640615446306 -alloc-id 10000
 ```
 
-### 重启整个集群
+### 第5 步：重启整个集群
```
Comment (lhy1024) [inline pd-recover.md:163]:
```suggestion
### 第 5 步：重启整个集群
```

File: pd-recover.md (hunk 3 above)
Comment (nolouch) [reply pd-recover.md:69]: 这里只介绍 pd-recovery, 可以考虑后面单独有一个恢复集群的文档。

**Comment by nolouch** [comment]
ptal @lhy1024 @Oreoxmt

**Review by lhy1024 — COMMENTED** [review]
the rest lgtm

File: pd-recover.md [hunk 5]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
 
 PD Recover 的安装包位于 TiDB 离线工具包中。下载方式，请参考 [TiDB 工具下载](/download-ecosystem-tools.md)。
 
-## 快速开始
+## 方式一：从剩余 PD 集群重建
```
Comment (Oreoxmt) [inline pd-recover.md:27]:
```suggestion
## 方式一：从存活的 PD 节点重建集群
```

File: pd-recover.md [hunk 6]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
 
-### 获取 Cluster ID
+PD 集群一旦超过大多数的节点发生灾难，则无法提供服务。当还有 PD 节点存活时，可以选择一个存活的 PD 节点强制修改 raft group 的成员为，从而使该节点恢复服务。具体方式为：
+
+### 第 1 步：停掉所有节点
```
Comment (Oreoxmt) [inline pd-recover.md:31]:
```suggestion
### 第 1 步：停止所有节点
```

File: pd-recover.md [hunk 7]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+把集群中的服务进程都停止 (TiKV/TiFlash/TiDB)，防止在还未恢复过程中与 PD 参数交互，造成数据错乱等无法挽救的异常状况。
```
Comment (Oreoxmt) [inline pd-recover.md:33]:
```suggestion
停止集群中的 TiDB、TiKV 和 TiFlash 服务进程，以防止在恢复过程中与 PD 参数交互，造成数据错乱或其他无法挽救的异常状况。
```

File: pd-recover.md [hunk 8]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+### 第 2 步：使用特定地启动参数拉起该存活的 PD 节点
```
Comment (Oreoxmt) [inline pd-recover.md:35]:
```suggestion
### 第 2 步：启动存活的 PD 节点
```

File: pd-recover.md [hunk 9]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+加上特定地启动参数 `--force-new-cluster` 拉起该存活的 PD 节点, 如：
```
Comment (Oreoxmt) [inline pd-recover.md:37]:
```suggestion
使用特定的启动参数 `--force-new-cluster` 拉起该存活的 PD 节点，如：
```

File: pd-recover.md [hunk 10]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+```
+ ./bin/pd-recover --from-old-member --endpoints=http://127.0.0.1:2379 // 指定对应的 PD 地址
+```
+
+> **注意：**
+>
+> 该步骤会自动将存储中的 alloc_id 加一个 safe guard 的值 100000000, 后续集群中分配的 ID 会偏大。
+>
+> 此外，pd-recover 不会对 tso 进行修改，请保证本地时间是大于故障发生时的时间的，且确认故障前 PD 组件之间是开启 NTP 时钟同步服务的。否则需要将本地时钟调整到一个比较未来的时间保障 TSO 不回退。
```
Comment (Oreoxmt) [inline pd-recover.md:59]:
```suggestion
> 该步骤会自动将存储中的 `alloc_id` 增加一个安全值 `100000000`，这样，后续集群中分配的 ID 会偏大。
>
> 此外，`pd-recover` 不会修改 TSO。因此，在执行此步骤之前请保证本地时间大于故障发生时间，并且确认故障前 PD 组件之间开启了 NTP 时钟同步服务。如果未开启，则需要将本地时钟调整到一个未来的时间保证 TSO 不会回退。
```

File: pd-recover.md [hunk 11]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+{{< copyable "shell-regular" >}}
+
+```
+ ./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml 
+```
+
+### 第 3 步： 使用 pd-recover 修复元数据
+
+由于是从少数派节点直接恢复服务，该节点可能存在数据是落后的，特别是 alloc_id，tso 这种一旦发生回退，则可能会给集群数据错乱或不可用的影响。因此，我们需要通过 pd-recover 来修改一些元数据，从而使得这个 PD 能继续提供正确的分配 ID，TSO 等服务，具体命令参考：
```
Comment (Oreoxmt) [inline pd-recover.md:47]:
```suggestion
由于该方法是利用少数派 PD 节点恢复服务，该节点可能存在数据落后的情况。对于 `alloc_id`、`tso` 这类数据，一旦发生回退，则可能会导致集群数据错乱或不可用。因此，需要使用 `pd-recover` 修改元数据，以确保该节点能提供正确的分配 ID、TSO 等服务。具体命令参考：
```

**Comment by qiancai** [comment]
Removed the **needs-cherry-pick-release-6.3** label because the v6.3 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v6.3 and will no longer receive new updates.

**Comment by Oreoxmt** [comment]
@nolouch PTAL

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-6.4 label because the v6.4 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v6.4 and will no longer receive new updates.

File: pd-recover.md [hunk 12]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+>
+> 此外，pd-recover 不会对 tso 进行修改，请保证本地时间是大于故障发生时的时间的，且确认故障前 PD 组件之间是开启 NTP 时钟同步服务的。否则需要将本地时钟调整到一个比较未来的时间保障 TSO 不回退。
+
+### 第 4 步：重启这个 PD
+
+当出现上一步出现 `recovery is successful` 的提示信息时，重启整个集群。
+
+### 第 5 步：扩容 PD，拉起集群
+
+通过部署工具扩容 PD，并把其他组件都拉起来。至此服务恢复
```
[the remark this answers is not in the document]
Comment (nolouch) [reply pd-recover.md:67]: 这个参考部署工具的文档吧，如 tiup 的

File: pd-recover.md [hunk 13]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+```
+ ./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml 
+```
+
+### 第 3 步：使用 `pd-recover` 修复元数据
+
+由于该方法是利用少数派 PD 节点恢复服务，该节点可能存在数据落后的情况。对于 `alloc_id`、`tso` 这类数据，一旦发生回退，则可能会导致集群数据错乱或不可用。因此，需要使用 `pd-recover` 修改元数据，以确保该节点能提供正确的分配 ID、TSO 等服务。具体命令参考：
+
+{{< copyable "shell-regular" >}}
+
```
Comment (Oreoxmt) [inline pd-recover.md:50]:
```suggestion

```

File: pd-recover.md [hunk 14]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+### 第 1 步：停止所有节点
+
+停止集群中的 TiDB、TiKV 和 TiFlash 服务进程，以防止在恢复过程中与 PD 参数交互，造成数据错乱或其他无法挽救的异常状况。
+
+### 第 2 步：启动存活的 PD 节点
+
+使用特定的启动参数 `--force-new-cluster` 拉起该存活的 PD 节点，如：
+
+{{< copyable "shell-regular" >}}
+
```
Comment (Oreoxmt) [inline pd-recover.md:40]:
```suggestion

```

File: pd-recover.md [hunk 15]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+```
```
Comment (Oreoxmt) [inline pd-recover.md:41]:
```suggestion
```shell
```

File: pd-recover.md [hunk 16]
```diff
@@ -24,9 +24,53 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+ ./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml 
+```
+
+### 第 3 步：使用 `pd-recover` 修复元数据
+
+由于该方法是利用少数派 PD 节点恢复服务，该节点可能存在数据落后的情况。对于 `alloc_id`、`tso` 这类数据，一旦发生回退，则可能会导致集群数据错乱或不可用。因此，需要使用 `pd-recover` 修改元数据，以确保该节点能提供正确的分配 ID、TSO 等服务。具体命令参考：
+
+{{< copyable "shell-regular" >}}
+
+```
```
Comment (Oreoxmt) [inline pd-recover.md:51]:
```suggestion
```shell
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Comment by Oreoxmt** [comment]
@lhy1024 PTAL again

File: pd-recover.md [hunk 17]
```diff
@@ -24,9 +24,47 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+### 第 2 步：启动存活的 PD 节点
+
+使用特定的启动参数 `--force-new-cluster` 拉起该存活的 PD 节点，如：
+```shell
+ ./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml 
+```
+
+### 第 3 步：使用 `pd-recover` 修复元数据
+
+由于该方法是利用少数派 PD 节点恢复服务，该节点可能存在数据落后的情况。对于 `alloc_id`、`tso` 这类数据，一旦发生回退，则可能会导致集群数据错乱或不可用。因此，需要使用 `pd-recover` 修改元数据，以确保该节点能提供正确的分配 ID、TSO 等服务。具体命令参考：
```
Comment (Oreoxmt) [inline pd-recover.md:44]:
```suggestion
由于该方法是利用少数派 PD 节点恢复服务，该节点可能存在数据落后的情况。对于 `alloc_id`、`tso` 这类数据，一旦发生回退，则可能会导致集群数据错乱或不可用。因此，需要使用 `pd-recover` 修改元数据，以确保该节点能提供正确的分配 ID、TSO 等服务。具体命令参考：

```

File: pd-recover.md [hunk 18]
```diff
@@ -24,9 +24,47 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
-### 获取 Cluster ID
+当 PD 集群的大多数节点发生灾难性故障时，集群将无法提供服务。当还有 PD 节点存活时，可以选择一个存活的 PD 节点，通过强制修改 Raft Group 的成员，使该节点重新恢复服务。具体操作步骤为：
+
+### 第 1 步：停止所有节点
+
+停止集群中的 TiDB、TiKV 和 TiFlash 服务进程，以防止在恢复过程中与 PD 参数交互，造成数据错乱或其他无法挽救的异常状况。
+
+### 第 2 步：启动存活的 PD 节点
+
+使用特定的启动参数 `--force-new-cluster` 拉起该存活的 PD 节点，如：
```
Comment (Oreoxmt) [inline pd-recover.md:37]:
```suggestion
使用特定的启动参数 `--force-new-cluster` 拉起该存活的 PD 节点，如：

```

File: pd-recover.md [hunk 19]
```diff
@@ -24,9 +24,49 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+> **注意：**
+>
+> 该步骤会自动将存储中的 `alloc_id` 增加一个安全值 `100000000`，这样，后续集群中分配的 ID 会偏大。
+>
+> 此外，`pd-recover` 不会修改 TSO。因此，在执行此步骤之前请保证本地时间大于故障发生时间，并且确认故障前 PD 组件之间开启了 NTP 时钟同步服务。如果未开启，则需要将本地时钟调整到一个未来的时间保证 TSO 不会回退。
+
+### 第 4 步：重启这个 PD
+
+当出现上一步出现 `recovery is successful` 的提示信息时，重启整个集群。
```
Comment (Oreoxmt) [inline pd-recover.md:59]:
```suggestion
当上一步出现 `recovery is successful` 的提示信息后，重启整个集群。
```
@nolouch 是重启整个集群还是这个 PD?

File: pd-recover.md [hunk 20]
```diff
@@ -24,9 +24,49 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+### 第 1 步：停止所有节点
+
+停止集群中的 TiDB、TiKV 和 TiFlash 服务进程，以防止在恢复过程中与 PD 参数交互，造成数据错乱或其他无法挽救的异常状况。
+
+### 第 2 步：启动存活的 PD 节点
+
+使用特定的启动参数 `--force-new-cluster` 拉起该存活的 PD 节点，如：
+
+```shell
+ ./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml
```
Comment (Oreoxmt) [inline pd-recover.md:40]:
```suggestion
./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml
```

File: pd-recover.md [hunk 21]
```diff
@@ -24,9 +24,49 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+```shell
+ ./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml 
+```
+
+### 第 3 步：使用 `pd-recover` 修复元数据
+
+由于该方法是利用少数派 PD 节点恢复服务，该节点可能存在数据落后的情况。对于 `alloc_id`、`tso` 这类数据，一旦发生回退，则可能会导致集群数据错乱或不可用。因此，需要使用 `pd-recover` 修改元数据，以确保该节点能提供正确的分配 ID、TSO 等服务。具体命令参考：
+
+```shell
+ ./bin/pd-recover --from-old-member --endpoints=http://127.0.0.1:2379 // 指定对应的 PD 地址
```
Comment (Oreoxmt) [inline pd-recover.md:48]:
```suggestion
./bin/pd-recover --from-old-member --endpoints=http://127.0.0.1:2379 // 指定对应的 PD 地址
```

File: pd-recover.md [hunk 22]
```diff
@@ -24,9 +24,49 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
 
 PD Recover 的安装包位于 TiDB 离线工具包中。下载方式，请参考 [TiDB 工具下载](/download-ecosystem-tools.md)。
```
Comment (Oreoxmt) [inline pd-recover.md:25]:
```suggestion
PD Recover 的安装包位于 TiDB 离线工具包中。下载方式，请参考 [TiDB 工具下载](/download-ecosystem-tools.md)。

下面介绍两种重建集群的方式：从存活的 PD 节点重建和完全重建。
```

File: pd-recover.md [hunk 23]
```diff
@@ -24,9 +24,49 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
-### 获取 Cluster ID
+当 PD 集群的大多数节点发生灾难性故障时，集群将无法提供服务。当还有 PD 节点存活时，可以选择一个存活的 PD 节点，通过强制修改 Raft Group 的成员，使该节点重新恢复服务。具体操作步骤为：
+
+### 第 1 步：停止所有节点
+
+停止集群中的 TiDB、TiKV 和 TiFlash 服务进程，以防止在恢复过程中与 PD 参数交互，造成数据错乱或其他无法挽救的异常状况。
+
+### 第 2 步：启动存活的 PD 节点
+
+使用特定的启动参数 `--force-new-cluster` 拉起该存活的 PD 节点，如：
```
Comment (Oreoxmt) [inline pd-recover.md:37]:
```suggestion
使用启动参数 `--force-new-cluster` 拉起该存活的 PD 节点，如：
```

File: pd-recover.md [hunk 24]
```diff
@@ -24,9 +24,49 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+```shell
+ ./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml 
+```
+
+### 第 3 步：使用 `pd-recover` 修复元数据
+
+由于该方法是利用少数派 PD 节点恢复服务，该节点可能存在数据落后的情况。对于 `alloc_id`、`tso` 这类数据，一旦发生回退，则可能会导致集群数据错乱或不可用。因此，需要使用 `pd-recover` 修改元数据，以确保该节点能提供正确的分配 ID、TSO 等服务。具体命令参考：
```
Comment (Oreoxmt) [inline pd-recover.md:45]:
```suggestion
由于该方法是利用少数派 PD 节点恢复服务，该节点可能存在数据落后的情况。对于 `alloc_id` 和 `tso` 等数据，一旦发生回退，可能导致集群数据错乱或不可用。因此，需要使用 `pd-recover` 修改元数据，以确保该节点能提供正确的分配 ID 和 TSO 等服务。具体命令参考：
```

File: pd-recover.md [hunk 25]
```diff
@@ -24,9 +24,49 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+```shell
+ ./bin/pd-recover --from-old-member --endpoints=http://127.0.0.1:2379 // 指定对应的 PD 地址
+```
+
+> **注意：**
+>
+> 该步骤会自动将存储中的 `alloc_id` 增加一个安全值 `100000000`，这样，后续集群中分配的 ID 会偏大。
+>
+> 此外，`pd-recover` 不会修改 TSO。因此，在执行此步骤之前请保证本地时间大于故障发生时间，并且确认故障前 PD 组件之间开启了 NTP 时钟同步服务。如果未开启，则需要将本地时钟调整到一个未来的时间保证 TSO 不会回退。
```
Comment (Oreoxmt) [inline pd-recover.md:55]:
```suggestion
> 该步骤会自动将存储中的 `alloc_id` 增加一个安全值 `100000000`。这将导致后续集群中分配的 ID 偏大。
>
> 此外，`pd-recover` 不会修改 TSO。因此，在执行此步骤之前，请确保本地时间大于故障发生时间，并且确认故障前 PD 组件之间已开启 NTP 时钟同步服务。如果未开启，则需要将本地时钟调整到一个未来的时间，以确保 TSO 不会回退。
```

File: pd-recover.md [hunk 26]
```diff
@@ -24,9 +24,49 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+### 第 4 步：重启这个 PD
+
+当出现上一步出现 `recovery is successful` 的提示信息时，重启整个集群。
+
+### 第 5 步：扩容 PD，拉起集群
+
+通过部署工具扩容 PD，并把其他组件都拉起来。至此服务恢复。
```
Comment (Oreoxmt) [inline pd-recover.md:63]:
```suggestion
### 第 5 步：扩容 PD 并启动集群

通过部署工具扩容 PD，并启动集群中的其他组件。至此服务恢复。
```

File: pd-recover.md [hunk 27]
```diff
@@ -24,9 +24,49 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+## 方式二：完全重建 PD 集群
+
+该方式适用于所有 PD 的数据都丢失，但其他组件数据都还存在的情况。
```
Comment (Oreoxmt) [inline pd-recover.md:67]:
```suggestion
该方式适用于所有 PD 的数据都丢失，但 TiDB、TiKV 和 TiFlash 等其他组件数据都还存在的情况。
```

File: pd-recover.md (hunk 19 above)
Comment (nolouch) [inline pd-recover.md]:
```suggestion
当出现上一步出现 `recovery is successful` 的提示信息时，重启 PD 。
```

Comment (nolouch) [inline pd-recover.md]:
```suggestion
当出现上一步出现 `recovery is successful` 的提示信息时，重启 PD。
```

File: pd-recover.md [hunk 28]
```diff
@@ -24,9 +24,51 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+> **注意：**
+>
+> 该步骤会自动将存储中的 `alloc_id` 增加一个安全值 `100000000`。这将导致后续集群中分配的 ID 偏大。
+>
+> 此外，`pd-recover` 不会修改 TSO。因此，在执行此步骤之前，请确保本地时间大于故障发生时间，并且确认故障前 PD 组件之间已开启 NTP 时钟同步服务。如果未开启，则需要将本地时钟调整到一个未来的时间，以确保 TSO 不会回退。
+
+### 第 4 步：重启这个 PD
+
+当上一步出现 `recovery is successful` 的提示信息后，重启整个集群。
```
Comment (nolouch) [inline pd-recover.md:61]:
```suggestion
当上一步出现 `recovery is successful` 的提示信息后，重启 PD。
```

**Comment by nolouch** [comment]
PTAL @Oreoxmt

File: pd-recover.md [hunk 29]
```diff
@@ -24,9 +24,51 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
 
 PD Recover 的安装包位于 TiDB 离线工具包中。下载方式，请参考 [TiDB 工具下载](/download-ecosystem-tools.md)。
 
-## 快速开始
+下面介绍两种重建集群的方式：从存活的 PD 节点重建和完全重建。
 
-### 获取 Cluster ID
+## 方式一：从存活的 PD 节点重建集群
+
+当 PD 集群的大多数节点发生灾难性故障时，集群将无法提供服务。当还有 PD 节点存活时，可以选择一个存活的 PD 节点，通过强制修改 Raft Group 的成员，使该节点重新恢复服务。具体操作步骤为：
```
Comment (ran-huang) [inline pd-recover.md:31]:
```suggestion
当 PD 集群的大多数节点发生灾难性故障时，集群将无法提供服务。当还有 PD 节点存活时，可以选择一个存活的 PD 节点，通过强制修改 Raft Group 的成员，使该节点重新恢复服务。具体操作步骤如下：
```

File: pd-recover.md [hunk 30]
```diff
@@ -24,9 +24,51 @@ PD Recover 是对 PD 进行灾难性恢复的工具，用于恢复无法正常
...
+
+使用启动参数 `--force-new-cluster` 拉起该存活的 PD 节点，如：
+
+```shell
+./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml
+```
+
+### 第 3 步：使用 `pd-recover` 修复元数据
+
+由于该方法是利用少数派 PD 节点恢复服务，该节点可能存在数据落后的情况。对于 `alloc_id` 和 `tso` 等数据，一旦发生回退，可能导致集群数据错乱或不可用。因此，需要使用 `pd-recover` 修改元数据，以确保该节点能提供正确的分配 ID 和 TSO 等服务。具体命令参考：
```
Comment (ran-huang) [inline pd-recover.md:47]:
```suggestion
该方法是利用少数派 PD 节点恢复服务，但由于该节点可能存在数据落后的情况，因此对于 `alloc_id` 和 `tso` 等数据，一旦发生回退，可能导致集群数据错乱或不可用。为确保该节点能提供正确的分配 ID 和 TSO 等服务，需要使用 `pd-recover` 修改元数据。具体命令参考：
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**